src/tinuous/github.py

```diff
@@ -98,11 +99,39 @@ def get_workflows(self) -> List[Workflow]:
     @retry_ratelimit
     def get_runs(self, wf: Workflow, since: datetime) -> List[WorkflowRun]:
         runs: List[WorkflowRun] = []
-        for r in wf.get_runs():
-            if ensure_aware(r.created_at) <= since:
-                break
-            runs.append(r)
-        return runs
+        try:
+            for r in wf.get_runs():
+                if ensure_aware(r.created_at) <= since:
+                    break
+                runs.append(r)
+            return runs
+        except requests.exceptions.RetryError as e:
+            reason = e.args[0].reason
+            if (
+                isinstance(reason, ResponseError)
+                and reason.args[0]
+                == ResponseError.SPECIFIC_ERROR.format(status_code=500)
```

Comment:
I will try now
```suggestion
                and (reason.args[0]
                 == ResponseError.SPECIFIC_ERROR.format(status_code=500)
                 or reason.args[0]
                 == ResponseError.SPECIFIC_ERROR.format(status_code=502))
```

Reply:
confirming that smth like that (I reentered manually) worked but needs adjustment of the log message below to log actual status_code - not hard-coded `500` -- please adjust accordingly and let's get it merged/released (adding a label).

Reply:
@yarikoptic Should the code just catch 500 and 502 here, or also 503 and 504 (the same as the status codes that are already retried)?

Reply:
Totally up to you - feel free to add those 5xx into the mix here as well .
- Originally I thought about it and because we are dealing with specific workaround, thought we better keep to 500 and 502 until we run into the case where any other 5xx shows up for this particular scenario since we do not want to skip any other similar case I guess.
- Since we do test for the `since` and presence of workflow file I think it would be ok if we skip on any 5xx.

Reply:
Updated: 74066c7